### PR TITLE
Port to Windows / MSVC Toolchain on Vulkan Branch

### PR DESCRIPTION
**Pull request by jmoguillansky-gpsw**
Porting vulkan branch to windows/msvc. 

Current status: Both OpenGL and Vulkan backends are working on windows platform with msvc toolchain.

Ok:                 421
Expected Fail:      0
Fail:               1

### CODE REVIEW

File: pynodegl/setup.py [hunk 1]
```diff
@@ -36,13 +36,18 @@ def __init__(self, pkg_config_bin='pkg-config'):
...
-        self.include_dirs = [f[2:] for f in flags if f.startswith('-I')]
-        self.library_dirs = [f[2:] for f in flags if f.startswith('-L')]
-        self.libraries    = [f[2:] for f in flags if f.startswith('-l')]
-
+        pkgcfg_cflags      = subprocess.check_output([pkg_config_bin, '--cflags', self.PKG_LIB_NAME]).decode()
+        pkgcfg_libs        = subprocess.check_output([pkg_config_bin, '--libs', self.PKG_LIB_NAME]).decode()
+        cflags = pkgcfg_cflags.split()
+        ldflags = pkgcfg_libs.split()
+        self.include_dirs = [f[2:] for f in cflags if f.startswith('-I')]
+        self.library_dirs = [f[2:] for f in ldflags if f.startswith('-L')]
```
Comment (cboesch-gpsw) [inline pynodegl/setup.py:44]: You want that change in a separate commit; is it fixing anything or is it just to make it more correct?

File: external/patches/sxplayer/sxplayer.patch [hunk 2]
```diff
@@ -1,40 +1,29 @@
...
 --- sxplayer-9.6.0/meson.build	2020-11-07 01:31:58.000000000 -0800
-+++ ../new/sxplayer-9.6.0/meson.build	2020-12-03 13:46:41.202305000 -0800
-@@ -58,6 +58,10 @@
-   dependency('libavutil', version: '>= 56.14.100'),
- ]
- 
-+if host_system == 'windows'  and cc.get_id() == 'msvc'
-+  lib_deps += dependency('libpthreads')
-+endif
-+
```
Comment (cboesch-gpsw) [inline external/patches/sxplayer/sxplayer.patch]: is this needed?

File: external/patches/sxplayer/sxplayer.patch [hunk 3]
```diff
@@ -1,40 +1,29 @@
...
- if vaapi_dep.found()
-   lib_deps += vaapi_dep
-@@ -102,6 +106,7 @@
++++ ../new/sxplayer-9.6.0/meson.build	2020-12-07 14:30:29.573943200 -0800
+@@ -102,6 +102,7 @@
    install: true,
    install_rpath: install_rpath,
    version: meson.project_version(),
-+  c_args: '-DBUILD_SXPLAYER_SHARED_LIB',
++  c_args: '-DBUILD_SXPLAYER',
```
Comment (cboesch-gpsw) [inline external/patches/sxplayer/sxplayer.patch]: mmh? I think you need to rebase your branch on top of master, it includes sxplayer 9.7.0

**Review by cboesch-gpsw — COMMENTED** [review]
Most of your changes should go away with a rebase on top of vulkan

File: pynodegl/setup.py (hunk 1 above)
Comment (jmoguillansky-gpsw) [reply pynodegl/setup.py:44]: This is related to adding logic to support windows libs.  I split libs and cflags into two separate variables.  I then split pkgcfg_libs and store it as ldflags variable.  Then, I iterate over ldflags and handle windows libs (ending with .lib).

File: external/patches/sxplayer/sxplayer.patch (hunk 2 above)
Comment (jmoguillansky-gpsw) [reply external/patches/sxplayer/sxplayer.patch]: We can remove this patch.  Currently in order to build sxplayer we need to use "vcpkg integrate install".  This is sort of an intrusive step: it changes the global visual studio configuration.  We may consider removing this dependency on "vcpkg integrate install" for sxplayer build.

**Comment by jmoguillansky-gpsw** [comment]
> Most of your changes should go away with a rebase on top of vulkan

Are we planning to first merge PR 152 into master and then this PR?  Maybe I can wait until 152 is merged, and then rebase this PR on top of the latest code related to the build system?

File: pynodegl/setup.py (hunk 1 above)
Comment (cboesch-gpsw) [reply pynodegl/setup.py:44]: I understood your code your intention, but it took me much longer than needed, which wouldn't have happened if that commit was split: first the split of cflags vs libs (which looks unrelated to the initial intention of adding .lib) and then the loop over .lib

**Comment by cboesch-gpsw** [comment]
> > Most of your changes should go away with a rebase on top of vulkan
> 
> Are we planning to first merge PR 152 into master and then this PR? Maybe I can wait until 152 is merged, and then rebase this PR on top of the latest code related to the build system?

ah so that's the one we were supposed to review first, ok

File: pynodegl/setup.py (hunk 1 above)
Comment (jmoguillansky-gpsw) [reply pynodegl/setup.py:44]: ok sure I'll add an extra commit

**Comment by jmoguillansky-gpsw** [comment]
moved to https://github.com/gopro/gopro-lib-node.gl/pull/233

**Pull request closed without merging**